### PR TITLE
Contracts: Test benchmarking v2

### PR DESCRIPTION
**Pull request by pgherveou**

### CODE REVIEW

**Comment by pgherveou** [comment]
bot help

File: substrate/frame/contracts/src/benchmarking/mod.rs [hunk 1]
```diff
@@ -16,7 +16,7 @@
 // limitations under the License.
 
 //! Benchmarks for the contracts pallet
-
+#![allow(unused)]
```
Comment (athei) [inline substrate/frame/contracts/src/benchmarking/mod.rs:19]: Can we put those per case instead of deactivating for the whole module?

**Review by athei — COMMENTED** [review]
Nice work :)

File: substrate/frame/contracts/src/benchmarking/mod.rs (hunk 1 above)
Comment (pgherveou) [reply substrate/frame/contracts/src/benchmarking/mod.rs:19]: Ah right that should not be needed that was just me experimenting with the new syntax

**Review by athei — APPROVED** [review]

**Comment by pgherveou** [comment]
Depends on https://github.com/paritytech/polkadot-sdk/pull/3574

File: substrate/frame/contracts/src/benchmarking/mod.rs [hunk 2]
```diff
@@ -1048,38 +1152,51 @@ benchmarks! {
...
 	}
 
-	seal_debug_message_per_byte {
-		// Vary size of input in bytes up to maximum allowed contract memory
-		// or maximum allowed debug buffer size, whichever is less.
-		let i in 0 .. (T::Schedule::get().limits.memory_pages * 64 * 1024).min(T::MaxDebugBufferLen::get());
+	// Vary size of input in bytes up to maximum allowed contract memory
+	// or maximum allowed debug buffer size, whichever is less.
+	#[benchmark]
+	#[rustfmt::skip]
```
Comment (pgherveou) [inline substrate/frame/contracts/src/benchmarking/mod.rs:1189]: can be removed once 
this is merged https://github.com/paritytech/polkadot-sdk/pull/3598

**Comment by pgherveou** [comment]
🔒 for now some of the generated pov are much larger
@ggwpez is working on adding pov_mode = measured

https://weights.tasty.limo/compare?repo=polkadot-sdk&threshold=10&path_pattern=substrate%2Fframe%2F**%2Fsrc%2Fweights.rs%2Cpolkadot%2Fruntime%2F*%2Fsrc%2Fweights%2F**%2F*.rs%2Cpolkadot%2Fbridges%2Fmodules%2F*%2Fsrc%2Fweights.rs%2Ccumulus%2F**%2Fweights%2F*.rs%2Ccumulus%2F**%2Fweights%2Fxcm%2F*.rs%2Ccumulus%2F**%2Fsrc%2Fweights.rs&method=asymptotic&ignore_errors=true&unit=proof&old=master&new=pg%2Fcontracts-benchmark-v2

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-clippy
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/5476378

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-clippy
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/5476947

**Comment by pgherveou** [comment]
bot bench substrate-pallet --pallet=pallet_contracts

**Comment by pgherveou** [comment]
bot bench substrate-pallet --pallet=pallet_contracts

**Comment by pgherveou** [comment]
bot bench substrate-pallet --pallet=pallet_contracts

**Review by xermicus — APPROVED** [review]

**Pull request merged**